### ISSUE TITLE
Feature: more color utilities

### ISSUE THREAD

**Issue by EricRovell** [omgovich/colord#61]
# More color utilities

## Summary

Add some more utility functions to work with color instances.

## Motivation

Some more simple utilities may be useful for the user:

1. `copy` or `clone`: provides an easy way to clone a color instance.
2. `equals`: provides an easy way to compare two colors.

## Detailed design

The are two possibilities as I see:

### As part of the interface

```js
import { colord } from "colord";

const color = colord("#DC3522");
const anotherColor = colord("#abc123");
const colorCopy = color.copy();

color.equals(colorCopy); // -> true
color.equals(anotherColor); // -> false;
```

### As utility functions

```js
import { colord, copy, equals } from "colord";

const color = colord("#DC3522");
const anotherColor = colord("#abc123");
const colorCopy = copy(color);

equals(colorCopy); // -> true
equals(anotherColor); // -> false;
```

## Discussion

I think it would be better for this utilities, if they are seems useful for you, to be part of the main interface. In any case, if you think this feature may be useful and should be implemented, I am ready to do this.

**Comment by omgovich** [comment]
Hi. In my opinion, since colord is 100% immutable (any action creates a new `Colord` instance) it doesn't make a lot of sense to implement `.copy()` method.

At the same time `equals` sounds good and shouldn't cost us many bytes.
But I would go with `color1.isEqual(color2)` or `color1.isSame(color2)` for better consistency.

**Comment by EricRovell** [comment]
Yeah, it makes sense. I will try to implement it this way then :)

**Comment by omgovich** [comment]
Just released v2.3.0. Thanks!
